embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/EmbraceApiService.kt

```diff
@@ -170,38 +172,45 @@ internal class EmbraceApiService(
         return postOnExecutor(bytes, request, onComplete)
     }
 
+    /**
+     * Posts an API call to the API.
+     * If the device is offline or the endpoint is rate limited, the API call is saved to be sent later.
+     * If the API call fails, it is retried and if it succeeds, it is removed from cache.
+     */
     private fun postOnExecutor(
         payload: ByteArray,
         request: ApiRequest,
         onComplete: (() -> Any)?
     ): Future<*> {
         return executorService.submit(
             NetworkRequestRunnable(request) {
-                try {
-                    if (lastNetworkStatus != NetworkStatus.NOT_REACHABLE) {
-                        executePost(request, payload)
+                val pendingApiCall = pendingApiCallsSender.savePendingApiCall(request, payload)
```

Comment:
The reason you are saving the pending call always is not directly related to the 429 work, but just to take care of the case when we crash during the a network call and we haven't persisted it, so we lose it?

Reply:
Yeap, but as we talked on slack we will address that on a separate task.